tests/test_tokenization_bartpho.py

```diff
@@ -0,0 +1,65 @@
+# coding=utf-8
+# Copyright 2020 HuggingFace Inc. team.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os
+import unittest
+
+from transformers.models.bartpho.tokenization_bartpho import VOCAB_FILES_NAMES, BartphoTokenizer
+
+from .test_tokenization_common import TokenizerTesterMixin
+
+
+SAMPLE_VOCAB = os.path.join(os.path.dirname(os.path.abspath(__file__)), "fixtures/test_sentencepiece_bpe.model")
+
+
+class BartphoTokenizerTest(TokenizerTesterMixin, unittest.TestCase):
+
+    tokenizer_class = BartphoTokenizer
+    test_rust_tokenizer = False
+
+    def setUp(self):
+        super().setUp()
+
+        vocab = ["▁This", "▁is", "▁a", "▁t", "est"]
+        vocab_tokens = dict(zip(vocab, range(len(vocab))))
+        self.special_tokens_map = {"unk_token": "<unk>"}
+
+        self.reduced_vocab_file = os.path.join(self.tmpdirname, VOCAB_FILES_NAMES["reduced_vocab_file"])
+        with open(self.reduced_vocab_file, "w", encoding="utf-8") as fp:
+            for token in vocab_tokens:
+                fp.write(f"{token} {vocab_tokens[token]}\n")
+
+        tokenizer = BartphoTokenizer(SAMPLE_VOCAB, self.reduced_vocab_file, **self.special_tokens_map)
+        tokenizer.save_pretrained(self.tmpdirname)
+
+    def get_tokenizer(self, **kwargs):
+        kwargs.update(self.special_tokens_map)
+        return BartphoTokenizer.from_pretrained(self.tmpdirname, **kwargs)
+
+    def get_input_output_texts(self, tokenizer):
+        input_text = "This is a là test"
+        output_text = "This is a<unk><unk> test"
+        return input_text, output_text
+
+    def test_full_tokenizer(self):
+        tokenizer = BartphoTokenizer(SAMPLE_VOCAB, self.reduced_vocab_file, **self.special_tokens_map)
+        text = "This is a là test"
+        bpe_tokens = "▁This ▁is ▁a ▁l à ▁t est".split()
+        tokens = tokenizer.tokenize(text)
+        self.assertListEqual(tokens, bpe_tokens)
+
+        input_tokens = tokens + [tokenizer.unk_token]
+        input_bpe_tokens = [4, 5, 6, 3, 3, 7, 8, 3]
```

Comment:
nice test! could we also add an integration test the main tokenizer, An example could found here

https://github.com/huggingface/transformers/blob/3a8a8013adc5802cc302e63e403f0e8925fab0ea/tests/test_tokenization_m2m_100.py#L112-L122

Reply:
Is this necessary? I am not sure.
This M2M100Tokenizer has an opposite operating manner compared to ours. M2M100Tokenizer allows adding extra types from new languages when continuing pre-training. 
Ours, on the other hand, wants to reduce the vocab size of an existing one (e.g. equivalent to reducing model sizes of XLM-RoBERTa or mBART when working with only Vietnamese).